data/transactions/verify/txn.go

```diff
@@ -188,11 +198,12 @@ func TxnGroup(stxs []transactions.SignedTxn, contextHdr bookkeeping.BlockHeader,
 
 // txnGroupBatchPrep verifies a []SignedTxn having no obviously inconsistent data.
 // it is the caller responsibility to call batchVerifier.Verify()
-func txnGroupBatchPrep(stxs []transactions.SignedTxn, contextHdr bookkeeping.BlockHeader, ledger logic.LedgerForSignature, verifier *crypto.BatchVerifier) (*GroupContext, error) {
+func txnGroupBatchPrep(stxs []transactions.SignedTxn, contextHdr bookkeeping.BlockHeader, ledger logic.LedgerForSignature, verifier *crypto.BatchVerifier, evalTracer logic.EvalTracer) (*GroupContext, error) {
```

Comment:
I'm a little surprised to see the Tracer show up in the groupctx, rather than the ep that evaluates the group.  Can you talk me through that?

(I'm worried because the group ctx also serves as a key describing when a logicsig must be re-evaluated. I haven't ever needed to look deeply at that code, but it has made me wary of changing the group ctx.)

Reply:
I think the thread raises a good question and prompts another question on my side:  If `GroupContext` retains `evalTracer`, can tests be added confirming `TxnGroupWithTracer` always supplies `nil` cache and `TxnGroup` always supplies a `nil` tracer?

From my brief review, it feels preferable to avoid editing `GroupContext` though I'm also interested in understanding the rationale.

Reply:
Thanks for brining up the concerns about modifying `GroupContext`. The only reason this was modified in the first place was to make it more convenient to access the tracer at the time the EvalParams are created for LogicSig evaluation.

In 10e74cb08, I've undone the changes to `GroupContext` in favor of adding the tracer as an additional argument to internal functions to achieve the same effect.